src/schema-parser/avro-schema-parser.ts

```diff
@@ -0,0 +1,347 @@
+import avsc, { type Schema} from "avsc";
+import type { AsyncAPISchema, SchemaValidateResult } from "types";
+import { SchemaParser, type ParseSchemaInput, type ValidateSchemaInput } from "../schema-parser";
+import { type JSONSchema7TypeName } from "json-schema";
+
+type AvroSchema = Schema & { [key: string]: any } & any;
+
+export function AvroSchemaParser(): SchemaParser {
+  return {
+    validate,
+    parse,
+    getMimeTypes,
+  }
+}
+
+async function validate(input: ValidateSchemaInput<unknown, unknown>): Promise<SchemaValidateResult[]> {
+  let result: SchemaValidateResult[] = []
+  
+  try {
+    validateAvroSchema(input.data as AvroSchema);  
+  } catch (error) {
+    // TODO try to parse errors: https://github.com/mtth/avsc/blob/c639014507f08dbbdffecae8ccae0118390d13f8/lib/types.js#L118
+    if (error instanceof Error) {
+      result.push({
+        message: error.message,
+        path: [], // TODO how to get the path?
+      });
+    }
+  }
+  
+  return result;
+}
+
+async function parse(input: ParseSchemaInput<unknown, unknown>): Promise<AsyncAPISchema> {
+  const asyncAPISchema = await avroToJsonSchema(input.data as AvroSchema);
+  const message = (input.meta as any).message
```

Comment:
The big change VS what we had previously is that the new interface passes directly the Schema in `input.data` rather than the message object. 
I decided to add the message object in the `input.meta` and only in case it is there, do the modifications (if they are being done here finally).

Reply:
Ok, I will remember about that :) Avro parser there is a bit problematic 😅